File: src/main/java/ru/vmsystems/template/domain/service/BackService.java
package ru.vmsystems.template.domain.service;

import org.springframework.beans.factory.annotation.Autowired;
import ru.vmsystems.template.domain.model.CompanyEntity;
import ru.vmsystems.template.domain.model.user.UserEntity;
import ru.vmsystems.template.infrastructure.persistence.UserRepository;

import javax.servlet.http.HttpServletRequest;
import java.util.Optional;

public abstract class BackService {

    @Autowired
    private HttpServletRequest httpRequest;
    @Autowired
    private UserRepository userRepository;
    @Autowired
    private ReceptionOfOrderService receptionOfOrderService;
    @Autowired
    private SessionService sessionService;

    public String getSesionId() {
        return httpRequest.getSession().getId();
    }

    public String getLogin() {
        return httpRequest.getRemoteUser();
    }

    public CompanyEntity getCompany() {
        Optional<UserEntity> user = userRepository.getByLogin(getLogin());
        if (user.isPresent()) {
            return user.get().getCompany();
        } else {
            throw new RuntimeException("user not found " + getLogin());
        }
    }

    public Long getCompanyId() {
        return getCompany().getId();
    }

    public String getReceptionOfOrderName() {
        return sessionService.getCurrentReceptionOfOrderName();
    }
}

File: src/main/java/ru/vmsystems/glasscloud/domain/BackService.kt
package ru.vmsystems.glasscloud.domain

import org.springframework.beans.factory.annotation.Autowired
import ru.vmsystems.glasscloud.domain.company.CompanyEntity
import ru.vmsystems.glasscloud.domain.company.CompanyRepository
import ru.vmsystems.glasscloud.domain.company.ReceptionService
import ru.vmsystems.glasscloud.user.UserRepository
import java.util.*
import javax.servlet.http.HttpServletRequest

abstract class BackService {
    @Autowired
    private lateinit var userRepository: UserRepository
    @Autowired
    private lateinit var companyRepository: CompanyRepository
    @Autowired
    private lateinit var receptionService: ReceptionService
    @Autowired
    private lateinit var sessionService: SessionService

    @Autowired
    private val httpRequest: HttpServletRequest? = null

    val sesionId: String
        get() = httpRequest!!.session.id

    val login: String
        get() = httpRequest!!.remoteUser

    val company: CompanyEntity
        get() {
            val user = userRepository.getByLogin(login) ?: throw RuntimeException("user not found $login")
            return companyRepository.getById(user.companyId!!) ?: throw RuntimeException("company not found $companyId")
        }

    val companyId: UUID
        get() = company.id!!

    val receptionName: String
        get() = sessionService.currentReceptionOfOrderName
}
